### PR TITLE
Added tone()

### PR DESCRIPTION
**Pull request by brentru**
Added tone() to generate square wave w/50% duty cycle and user-defined frequency. #2

### CODE REVIEW

File: simpleio.py [hunk 1]
```diff
@@ -32,8 +32,96 @@
...
+
+
+def noTone(pin):
+    """
+    Stops the generation of the square wave produced by tone()
+    by deinit'ing the pin.
+
+    :param pin: pin on which to stop the tone from being played
+    """
+    pwm.deinit(pin)
```
Comment (tannewt) [inline simpleio.py]: Have you tried this? I don't think the pwm object will persist across the `tone` and `noTone` calls. It also doesn't take a pin in.

Adding support for the analog out pin would also be nice. PWM doesn't work with it but AudioOut does.

File: simpleio.py [hunk 2]
```diff
@@ -32,8 +32,96 @@
...
+
+def shiftIn(dataPin, clock, bitOrder):
+    """
+    Shifts in a byte of data one bit at a time. Starts from either the LSB or
+    MSB.
+
+    :param dataPin: pin on which to input each bit
+    :param clock: toggles to signal dataPin reads
+    :param bitOrder: order to shift bits (either LSBFIRST or MSBFIRST)
```
Comment (tannewt) [inline simpleio.py]: I'd make this a bool instead of a string and give it a default. So something like `msb_first=True`.

File: simpleio.py [hunk 3]
```diff
@@ -32,8 +32,96 @@
...
+    :param value: RETURNS the value read
```
Comment (tannewt) [inline simpleio.py]: Returns are documented like:

```
:return: the device's usage page
:rtype: int
```

(from [here](https://github.com/adafruit/circuitpython/blob/master/shared-bindings/usb_hid/Device.c#L72))

File: simpleio.py [hunk 4]
```diff
@@ -32,8 +32,96 @@
...
+
+    .. code-block:: python
+
+        import digitalio
+        import simpleio
+        from board import *
+        clock = digitalio.DigitalInOut(D12)
+        dataPin = digitalio.DigitalInOut(D11)
+        clock.switch_to_output()
+        dataPin.switch_to_output()
```
Comment (tannewt) [inline simpleio.py]: Lets use the property in examples where we don't provide arguments to `switch_to_output`.

So it should be `clock.direction = digitalio.Direction.OUTPUT`

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for picking these up!

**Comment by tannewt** [comment]
Please rebase this now that the shift code is in.

**Comment by brentru** [comment]
Rebased.

**Comment by tannewt** [comment]
What did you rebase onto? It looks likes the tone branch is still based on a commit from April. The `Files changed` tab shows a ton of changes unrelated to `tone`.

File: simpleio.py [hunk 5]
```diff
@@ -25,12 +25,39 @@
...
+            else:
+                square_wave.append(0x00)
+        with audioio.AudioOut(pin, square_wave) as waveform:
+            waveform.play(loop=True)
+            time.sleep(duration)
+            waveform.stop()
+    except ValueError:
+        with pulseio.PWMOut(pin, frequency = frequency, variable_frequency = False) as pwm:
+            pwm.duty_cycle = 0x8000
+            time.sleep(1)
```
Comment (tannewt) [inline simpleio.py]: `time.sleep(duration)`

**Review by tannewt — APPROVED** [review]
Thank you so much for doing this!

**Pull request merged**